[Build Speed] Remove ContextDestructionObserverInlines.h from all non-Inlines.h headers

jernoble opened the pull request:
#### 326e827ddfb9a016d2a09256e8ed8983e37d53fc
<pre>
[Build Speed] Remove ContextDestructionObserverInlines.h from all non-Inlines.h headers
<a href="https://bugs.webkit.org/show_bug.cgi?id=298357">https://bugs.webkit.org/show_bug.cgi?id=298357</a>
<a href="https://rdar.apple.com/159802483">rdar://159802483</a>

Reviewed by Darin Adler.

Prior to this patch, there were 12 instances of including ContextDestructionObserverInlines.h in
non-Inlines.h headers, the most of any WebKit Inlines.h header.

Most of the uses of ContextDestructionObserverInlines.h come from implementing the virtual
scriptExecutionContext() method. Because this method is virtual, there&apos;s no benefit from it being
inlined in a header, so the implementation of these methods were moved into cpp files.

In other cases, the use of Inlines.h methods were themselves used in inlined methods. Those methods
were moved to new Inlines.h headers, and callers were updated to include those new header files.

These changes exposed a number of instances where files which used these inlined functions did not
include the correct Inlines.h header.

After this patch, zero instances remain of ContextDestructionObserverInlines.h in non-Inlines.h
headers.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/applepay-ams-ui/ApplePayAMSUIPaymentHandler.cpp:
* Source/WebCore/Modules/applepay/ApplePaySession.cpp:
(WebCore::ApplePaySession::scriptExecutionContext const):
* Source/WebCore/Modules/applepay/ApplePaySession.h:
* Source/WebCore/Modules/applepay/ApplePaySetup.cpp:
* Source/WebCore/Modules/async-clipboard/Clipboard.cpp:
* Source/WebCore/Modules/async-clipboard/ClipboardItem.cpp:
* Source/WebCore/Modules/audiosession/NavigatorAudioSession.cpp:
* Source/WebCore/Modules/cache/DOMCache.cpp:
* Source/WebCore/Modules/cookie-store/CookieStore.cpp:
* Source/WebCore/Modules/encryptedmedia/MediaKeySession.cpp:
(WebCore::MediaKeySession::scriptExecutionContext const):
* Source/WebCore/Modules/encryptedmedia/MediaKeySession.h:
* Source/WebCore/Modules/encryptedmedia/MediaKeySystemRequest.cpp:
* Source/WebCore/Modules/encryptedmedia/legacy/WebKitMediaKeySession.cpp:
(WebCore::WebKitMediaKeySession::scriptExecutionContext const):
* Source/WebCore/Modules/encryptedmedia/legacy/WebKitMediaKeySession.h:
* Source/WebCore/Modules/fetch/FetchRequest.cpp:
* Source/WebCore/Modules/geolocation/Geolocation.cpp:
* Source/WebCore/Modules/indexeddb/IDBActiveDOMObject.h:
(WebCore::IDBActiveDOMObject::performCallbackOnOriginThread): Deleted.
(WebCore::IDBActiveDOMObject::callFunctionOnOriginThread): Deleted.
* Source/WebCore/Modules/indexeddb/IDBActiveDOMObjectInlines.h:
(WebCore::IDBActiveDOMObject::performCallbackOnOriginThread):
(WebCore::IDBActiveDOMObject::callFunctionOnOriginThread):
* Source/WebCore/Modules/indexeddb/IDBDatabase.cpp:
(WebCore::IDBDatabase::scriptExecutionContext const):
* Source/WebCore/Modules/indexeddb/IDBDatabase.h:
* Source/WebCore/Modules/indexeddb/IDBIndex.cpp:
* Source/WebCore/Modules/indexeddb/IDBRequest.cpp:
(WebCore::IDBRequest::scriptExecutionContext const):
* Source/WebCore/Modules/indexeddb/IDBRequest.h:
* Source/WebCore/Modules/indexeddb/IDBTransaction.cpp:
(WebCore::IDBTransaction::scriptExecutionContext const):
* Source/WebCore/Modules/indexeddb/IDBTransaction.h:
* Source/WebCore/Modules/indexeddb/client/IDBConnectionProxy.cpp:
* Source/WebCore/Modules/indexeddb/client/IDBConnectionProxy.h:
* Source/WebCore/Modules/indexeddb/client/TransactionOperation.cpp:
(WebCore::IDBClient::TransactionOperation::transitionToComplete):
* Source/WebCore/Modules/indexeddb/client/TransactionOperation.h:
(WebCore::IDBClient::TransactionOperation::transitionToComplete): Deleted.
* Source/WebCore/Modules/mediacontrols/MediaControlsUtils.cpp:
* Source/WebCore/Modules/mediarecorder/MediaRecorder.cpp:
(WebCore::MediaRecorder::scriptExecutionContext const):
* Source/WebCore/Modules/mediarecorder/MediaRecorder.h:
* Source/WebCore/Modules/mediasession/MediaSession.cpp:
* Source/WebCore/Modules/mediasession/MediaSessionCoordinator.cpp:
(WebCore::MediaSessionCoordinator::scriptExecutionContext const):
* Source/WebCore/Modules/mediasession/MediaSessionCoordinator.h:
* Source/WebCore/Modules/mediasource/MediaSource.cpp:
* Source/WebCore/Modules/mediasource/SourceBufferList.cpp:
(WebCore::SourceBufferList::scriptExecutionContext const):
* Source/WebCore/Modules/mediasource/SourceBufferList.h:
* Source/WebCore/Modules/mediastream/RTCDTMFSender.cpp:
(WebCore::RTCDTMFSender::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/RTCDTMFSender.h:
* Source/WebCore/Modules/mediastream/RTCIceTransport.cpp:
(WebCore::RTCIceTransport::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/RTCIceTransport.h:
* Source/WebCore/Modules/mediastream/RTCPeerConnection.cpp:
(WebCore::RTCPeerConnection::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/RTCPeerConnection.h:
* Source/WebCore/Modules/mediastream/RTCSctpTransport.cpp:
(WebCore::RTCSctpTransport::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/RTCSctpTransport.h:
* Source/WebCore/Modules/mediastream/UserMediaRequest.cpp:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCMediaEndpoint.cpp:
(WebCore::LibWebRTCMediaEndpoint::addTrack):
(WebCore::LibWebRTCMediaEndpoint::createSourceAndRTCTrack):
(WebCore::LibWebRTCMediaEndpoint::mediaStreamsFromRTCStreamIds):
* Source/WebCore/Modules/notifications/Notification.cpp:
(WebCore::Notification::scriptExecutionContext const):
* Source/WebCore/Modules/notifications/Notification.h:
* Source/WebCore/Modules/notifications/NotificationResourcesLoader.cpp:
* Source/WebCore/Modules/paymentrequest/PaymentRequest.cpp:
(WebCore::PaymentRequest::scriptExecutionContext const):
* Source/WebCore/Modules/paymentrequest/PaymentRequest.h:
* Source/WebCore/Modules/paymentrequest/PaymentResponse.cpp:
(WebCore::PaymentResponse::scriptExecutionContext const):
* Source/WebCore/Modules/paymentrequest/PaymentResponse.h:
* Source/WebCore/Modules/pictureinpicture/PictureInPictureWindow.cpp:
(WebCore::PictureInPictureWindow::scriptExecutionContext const):
* Source/WebCore/Modules/pictureinpicture/PictureInPictureWindow.h:
* Source/WebCore/Modules/remoteplayback/RemotePlayback.cpp:
(WebCore::RemotePlayback::scriptExecutionContext const):
* Source/WebCore/Modules/remoteplayback/RemotePlayback.h:
* Source/WebCore/Modules/screen-wake-lock/WakeLockSentinel.cpp:
(WebCore::WakeLockSentinel::scriptExecutionContext const):
* Source/WebCore/Modules/screen-wake-lock/WakeLockSentinel.h:
* Source/WebCore/Modules/speech/SpeechRecognition.cpp:
(WebCore::SpeechRecognition::scriptExecutionContext const):
* Source/WebCore/Modules/speech/SpeechRecognition.h:
* Source/WebCore/Modules/speech/SpeechSynthesis.cpp:
(WebCore::SpeechSynthesis::scriptExecutionContext const):
* Source/WebCore/Modules/speech/SpeechSynthesis.h:
* Source/WebCore/Modules/web-locks/WebLockManager.cpp:
* Source/WebCore/Modules/webaudio/BaseAudioContext.cpp:
* Source/WebCore/Modules/websockets/WebSocket.cpp:
* Source/WebCore/Modules/webxr/WebXRSystem.cpp:
(WebCore::WebXRSystem::scriptExecutionContext const):
* Source/WebCore/Modules/webxr/WebXRSystem.h:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/animation/WebAnimation.cpp:
(WebCore::WebAnimation::scriptExecutionContext const):
* Source/WebCore/animation/WebAnimation.h:
* Source/WebCore/bindings/js/JSCustomElementInterface.cpp:
(WebCore::JSCustomElementInterface::scriptExecutionContext const):
* Source/WebCore/bindings/js/JSCustomElementInterface.h:
(WebCore::JSCustomElementInterface::scriptExecutionContext const): Deleted.
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.cpp:
* Source/WebCore/css/CSSFontFace.cpp:
* Source/WebCore/css/FontFaceSet.cpp:
(WebCore::FontFaceSet::scriptExecutionContext const):
* Source/WebCore/css/FontFaceSet.h:
* Source/WebCore/css/MediaQueryList.cpp:
(WebCore::MediaQueryList::scriptExecutionContext const):
* Source/WebCore/css/MediaQueryList.h:
* Source/WebCore/dom/AbortSignal.cpp:
(WebCore::AbortSignal::scriptExecutionContext const):
* Source/WebCore/dom/AbortSignal.h:
* Source/WebCore/dom/BroadcastChannel.cpp:
(WebCore::BroadcastChannel::scriptExecutionContext const):
* Source/WebCore/dom/BroadcastChannel.h:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/EventTargetConcrete.cpp:
(WebCore::EventTargetConcrete::scriptExecutionContext const):
* Source/WebCore/dom/EventTargetConcrete.h:
* Source/WebCore/dom/InternalObserverForEach.cpp:
* Source/WebCore/dom/InternalObserverInspect.cpp:
* Source/WebCore/dom/MessagePort.cpp:
(WebCore::MessagePort::scriptExecutionContext const):
* Source/WebCore/dom/MessagePort.h:
* Source/WebCore/dom/MutationObserver.cpp:
* Source/WebCore/dom/ScriptExecutionContext.cpp:
* Source/WebCore/dom/Subscriber.cpp:
* Source/WebCore/fileapi/FileReader.cpp:
(WebCore::FileReader::scriptExecutionContext const):
* Source/WebCore/fileapi/FileReader.h:
* Source/WebCore/html/HTMLMediaElement.cpp:
* Source/WebCore/html/ImageBitmap.cpp:
* Source/WebCore/html/OffscreenCanvas.cpp:
(WebCore::OffscreenCanvas::scriptExecutionContext const):
(WebCore::OffscreenCanvas::canvasBaseScriptExecutionContext const):
* Source/WebCore/html/OffscreenCanvas.h:
* Source/WebCore/html/VoidCallback.h:
* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
* Source/WebCore/html/closewatcher/CloseWatcher.cpp:
(WebCore::CloseWatcher::CloseWatcher):
(WebCore::CloseWatcher::scriptExecutionContext const):
* Source/WebCore/html/closewatcher/CloseWatcher.h:
* Source/WebCore/html/track/TextTrack.cpp:
(WebCore::TextTrack::scriptExecutionContext const):
* Source/WebCore/html/track/TextTrack.h:
* Source/WebCore/html/track/TrackListBase.cpp:
* Source/WebCore/inspector/InspectorInstrumentation.cpp:
(WebCore::InspectorInstrumentation::willDestroyWebGLProgram):
* Source/WebCore/inspector/InspectorInstrumentation.h:
(WebCore::InspectorInstrumentation::willDestroyWebGLProgram): Deleted.
* Source/WebCore/inspector/RTCLogsCallback.h:
* Source/WebCore/page/DOMTimer.cpp:
* Source/WebCore/page/EventSource.cpp:
(WebCore::EventSource::scriptExecutionContext const):
* Source/WebCore/page/EventSource.h:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::scriptExecutionContext const):
* Source/WebCore/page/LocalDOMWindow.h:
* Source/WebCore/page/NavigationHistoryEntry.cpp:
* Source/WebCore/page/NavigatorBase.cpp:
* Source/WebCore/page/Page.cpp:
* Source/WebCore/page/PerformanceObserver.cpp:
* Source/WebCore/page/ScreenOrientation.cpp:
(WebCore::ScreenOrientation::lock):
* Source/WebCore/page/ScreenOrientation.h:
* Source/WebCore/page/WorkerNavigator.cpp:
* Source/WebCore/testing/EventTargetForTesting.cpp:
(WebCore::EventTargetForTesting::scriptExecutionContext const):
* Source/WebCore/testing/EventTargetForTesting.h:
* Source/WebCore/testing/Internals.cpp:
* Source/WebCore/workers/Worker.cpp:
(WebCore::Worker::scriptExecutionContext const):
* Source/WebCore/workers/Worker.h:
* Source/WebCore/workers/WorkerGlobalScope.cpp:
* Source/WebCore/workers/service/ServiceWorkerContainer.cpp:
(WebCore::ServiceWorkerContainer::scriptExecutionContext const):
* Source/WebCore/workers/service/ServiceWorkerContainer.h:
* Source/WebCore/workers/service/ServiceWorkerRegistration.cpp:
* Source/WebCore/workers/shared/SharedWorker.cpp:
* Source/WebCore/workers/shared/SharedWorkerScriptLoader.cpp:
* Source/WebCore/worklets/Worklet.cpp:
* Source/WebCore/xml/XMLHttpRequest.cpp:
(WebCore::XMLHttpRequest::scriptExecutionContext const):
* Source/WebCore/xml/XMLHttpRequest.h:
* Source/WebCore/xml/XMLHttpRequestUpload.cpp:
(WebCore::XMLHttpRequestUpload::scriptExecutionContext const):
* Source/WebCore/xml/XMLHttpRequestUpload.h:

Canonical link: <a href="https://commits.webkit.org/299694@main">https://commits.webkit.org/299694@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fdb65d0ac15c2db17b9a045ae5c996c05e0ce8e0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/119921 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/39614 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30265 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/126239 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/71994 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/f2311f14-aa77-428c-9a3b-78e6233a971e) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/121797 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40309 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/48191 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91063 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60377 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/f7bd8bc5-5f28-4bec-8661-521c69e90828) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/122873 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32192 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107527 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71620 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/cea68a38-406c-45e8-b15c-7722d0929612) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31222 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25633 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/69882 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101665 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25823 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129167 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/46841 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/35509 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99686 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47207 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103718 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99531 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25259 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/44981 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/22995 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/43458 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46703 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/52409 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46169 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/49518 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/47855 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/Modules/indexeddb/client/TransactionOperationInlines.h:
```diff
@@ -0,0 +1,50 @@
+
+namespace IDBClient {
+
+void TransactionOperation::transitionToComplete(const IDBResultData& data, RefPtr<TransactionOperation>&& lastRef)
```
Maybe this function could just not be inlined instead of adding a new inlines header?

darinadler commented on Source/WebCore/Modules/indexeddb/IDBActiveDOMObjectInlines.h:
```diff
@@ -0,0 +1,69 @@
+    context->postCrossThreadTask(object, method, arguments...);
+}
+
+inline void IDBActiveDOMObject::callFunctionOnOriginThread(Function<void()>&& function)
```
Seems like this function probably just shouldn’t be inlined.

jernoble replied:
This class was originally a single-header-only one, so no .cpp file exists for this method to be placed in. Either way, it involves adding a new file to the project, and there needs to be a inlines.h file for the template definition anyway, so this seemed the best choice.

darinadler replied:
> This class was originally a single-header-only one, so no .cpp file exists for this method to be placed in.

Understood. In future I’m sure it would be OK to move this to a `.cpp` file but little harm for now in compiling it a few extra times.

darinadler commented on Source/WebCore/inspector/InspectorInstrumentationInlines.h:
```diff
@@ -0,0 +1,41 @@
+
+namespace WebCore {
+
+inline void InspectorInstrumentation::willDestroyWebGLProgram(WebGLProgram& program)
```
This function doesn’t need to be inlined.

jernoble committed:
[Build Speed] Remove ContextDestructionObserverInlines.h from all non-Inlines.h headers
https://bugs.webkit.org/show_bug.cgi?id=298357
rdar://159802483

Reviewed by Darin Adler.

Prior to this patch, there were 12 instances of including ContextDestructionObserverInlines.h in
non-Inlines.h headers, the most of any WebKit Inlines.h header.

Most of the uses of ContextDestructionObserverInlines.h come from implementing the virtual
scriptExecutionContext() method. Because this method is virtual, there's no benefit from it being
inlined in a header, so the implementation of these methods were moved into cpp files.

In other cases, the use of Inlines.h methods were themselves used in inlined methods. Those methods
were moved to new Inlines.h headers, and callers were updated to include those new header files.

These changes exposed a number of instances where files which used these inlined functions did not
include the correct Inlines.h header.

After this patch, zero instances remain of ContextDestructionObserverInlines.h in non-Inlines.h
headers.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/applepay-ams-ui/ApplePayAMSUIPaymentHandler.cpp:
* Source/WebCore/Modules/applepay/ApplePaySession.cpp:
(WebCore::ApplePaySession::scriptExecutionContext const):
* Source/WebCore/Modules/applepay/ApplePaySession.h:
* Source/WebCore/Modules/applepay/ApplePaySetup.cpp:
* Source/WebCore/Modules/async-clipboard/Clipboard.cpp:
* Source/WebCore/Modules/async-clipboard/ClipboardItem.cpp:
* Source/WebCore/Modules/audiosession/NavigatorAudioSession.cpp:
* Source/WebCore/Modules/cache/DOMCache.cpp:
* Source/WebCore/Modules/cookie-store/CookieStore.cpp:
* Source/WebCore/Modules/encryptedmedia/MediaKeySession.cpp:
(WebCore::MediaKeySession::scriptExecutionContext const):
* Source/WebCore/Modules/encryptedmedia/MediaKeySession.h:
* Source/WebCore/Modules/encryptedmedia/MediaKeySystemRequest.cpp:
* Source/WebCore/Modules/encryptedmedia/legacy/WebKitMediaKeySession.cpp:
(WebCore::WebKitMediaKeySession::scriptExecutionContext const):
* Source/WebCore/Modules/encryptedmedia/legacy/WebKitMediaKeySession.h:
* Source/WebCore/Modules/fetch/FetchRequest.cpp:
* Source/WebCore/Modules/geolocation/Geolocation.cpp:
* Source/WebCore/Modules/indexeddb/IDBActiveDOMObject.h:
(WebCore::IDBActiveDOMObject::performCallbackOnOriginThread): Deleted.
(WebCore::IDBActiveDOMObject::callFunctionOnOriginThread): Deleted.
* Source/WebCore/Modules/indexeddb/IDBActiveDOMObjectInlines.h:
(WebCore::IDBActiveDOMObject::performCallbackOnOriginThread):
(WebCore::IDBActiveDOMObject::callFunctionOnOriginThread):
* Source/WebCore/Modules/indexeddb/IDBDatabase.cpp:
(WebCore::IDBDatabase::scriptExecutionContext const):
* Source/WebCore/Modules/indexeddb/IDBDatabase.h:
* Source/WebCore/Modules/indexeddb/IDBIndex.cpp:
* Source/WebCore/Modules/indexeddb/IDBRequest.cpp:
(WebCore::IDBRequest::scriptExecutionContext const):
* Source/WebCore/Modules/indexeddb/IDBRequest.h:
* Source/WebCore/Modules/indexeddb/IDBTransaction.cpp:
(WebCore::IDBTransaction::scriptExecutionContext const):
* Source/WebCore/Modules/indexeddb/IDBTransaction.h:
* Source/WebCore/Modules/indexeddb/client/IDBConnectionProxy.cpp:
* Source/WebCore/Modules/indexeddb/client/IDBConnectionProxy.h:
* Source/WebCore/Modules/indexeddb/client/TransactionOperation.cpp:
(WebCore::IDBClient::TransactionOperation::transitionToComplete):
* Source/WebCore/Modules/indexeddb/client/TransactionOperation.h:
(WebCore::IDBClient::TransactionOperation::transitionToComplete): Deleted.
* Source/WebCore/Modules/mediacontrols/MediaControlsUtils.cpp:
* Source/WebCore/Modules/mediarecorder/MediaRecorder.cpp:
(WebCore::MediaRecorder::scriptExecutionContext const):
* Source/WebCore/Modules/mediarecorder/MediaRecorder.h:
* Source/WebCore/Modules/mediasession/MediaSession.cpp:
* Source/WebCore/Modules/mediasession/MediaSessionCoordinator.cpp:
(WebCore::MediaSessionCoordinator::scriptExecutionContext const):
* Source/WebCore/Modules/mediasession/MediaSessionCoordinator.h:
* Source/WebCore/Modules/mediasource/MediaSource.cpp:
* Source/WebCore/Modules/mediasource/SourceBufferList.cpp:
(WebCore::SourceBufferList::scriptExecutionContext const):
* Source/WebCore/Modules/mediasource/SourceBufferList.h:
* Source/WebCore/Modules/mediastream/RTCDTMFSender.cpp:
(WebCore::RTCDTMFSender::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/RTCDTMFSender.h:
* Source/WebCore/Modules/mediastream/RTCIceTransport.cpp:
(WebCore::RTCIceTransport::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/RTCIceTransport.h:
* Source/WebCore/Modules/mediastream/RTCPeerConnection.cpp:
(WebCore::RTCPeerConnection::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/RTCPeerConnection.h:
* Source/WebCore/Modules/mediastream/RTCSctpTransport.cpp:
(WebCore::RTCSctpTransport::scriptExecutionContext const):
* Source/WebCore/Modules/mediastream/RTCSctpTransport.h:
* Source/WebCore/Modules/mediastream/UserMediaRequest.cpp:
* Source/WebCore/Modules/mediastream/libwebrtc/LibWebRTCMediaEndpoint.cpp:
(WebCore::LibWebRTCMediaEndpoint::addTrack):
(WebCore::LibWebRTCMediaEndpoint::createSourceAndRTCTrack):
(WebCore::LibWebRTCMediaEndpoint::mediaStreamsFromRTCStreamIds):
* Source/WebCore/Modules/notifications/Notification.cpp:
(WebCore::Notification::scriptExecutionContext const):
* Source/WebCore/Modules/notifications/Notification.h:
* Source/WebCore/Modules/notifications/NotificationResourcesLoader.cpp:
* Source/WebCore/Modules/paymentrequest/PaymentRequest.cpp:
(WebCore::PaymentRequest::scriptExecutionContext const):
* Source/WebCore/Modules/paymentrequest/PaymentRequest.h:
* Source/WebCore/Modules/paymentrequest/PaymentResponse.cpp:
(WebCore::PaymentResponse::scriptExecutionContext const):
* Source/WebCore/Modules/paymentrequest/PaymentResponse.h:
* Source/WebCore/Modules/pictureinpicture/PictureInPictureWindow.cpp:
(WebCore::PictureInPictureWindow::scriptExecutionContext const):
* Source/WebCore/Modules/pictureinpicture/PictureInPictureWindow.h:
* Source/WebCore/Modules/remoteplayback/RemotePlayback.cpp:
(WebCore::RemotePlayback::scriptExecutionContext const):
* Source/WebCore/Modules/remoteplayback/RemotePlayback.h:
* Source/WebCore/Modules/screen-wake-lock/WakeLockSentinel.cpp:
(WebCore::WakeLockSentinel::scriptExecutionContext const):
* Source/WebCore/Modules/screen-wake-lock/WakeLockSentinel.h:
* Source/WebCore/Modules/speech/SpeechRecognition.cpp:
(WebCore::SpeechRecognition::scriptExecutionContext const):
* Source/WebCore/Modules/speech/SpeechRecognition.h:
* Source/WebCore/Modules/speech/SpeechSynthesis.cpp:
(WebCore::SpeechSynthesis::scriptExecutionContext const):
* Source/WebCore/Modules/speech/SpeechSynthesis.h:
* Source/WebCore/Modules/web-locks/WebLockManager.cpp:
* Source/WebCore/Modules/webaudio/BaseAudioContext.cpp:
* Source/WebCore/Modules/websockets/WebSocket.cpp:
* Source/WebCore/Modules/webxr/WebXRSystem.cpp:
(WebCore::WebXRSystem::scriptExecutionContext const):
* Source/WebCore/Modules/webxr/WebXRSystem.h:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/animation/WebAnimation.cpp:
(WebCore::WebAnimation::scriptExecutionContext const):
* Source/WebCore/animation/WebAnimation.h:
* Source/WebCore/bindings/js/JSCustomElementInterface.cpp:
(WebCore::JSCustomElementInterface::scriptExecutionContext const):
* Source/WebCore/bindings/js/JSCustomElementInterface.h:
(WebCore::JSCustomElementInterface::scriptExecutionContext const): Deleted.
* Source/WebCore/bindings/js/JSDOMPromiseDeferred.cpp:
* Source/WebCore/css/CSSFontFace.cpp:
* Source/WebCore/css/FontFaceSet.cpp:
(WebCore::FontFaceSet::scriptExecutionContext const):
* Source/WebCore/css/FontFaceSet.h:
* Source/WebCore/css/MediaQueryList.cpp:
(WebCore::MediaQueryList::scriptExecutionContext const):
* Source/WebCore/css/MediaQueryList.h:
* Source/WebCore/dom/AbortSignal.cpp:
(WebCore::AbortSignal::scriptExecutionContext const):
* Source/WebCore/dom/AbortSignal.h:
* Source/WebCore/dom/BroadcastChannel.cpp:
(WebCore::BroadcastChannel::scriptExecutionContext const):
* Source/WebCore/dom/BroadcastChannel.h:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/EventTargetConcrete.cpp:
(WebCore::EventTargetConcrete::scriptExecutionContext const):
* Source/WebCore/dom/EventTargetConcrete.h:
* Source/WebCore/dom/InternalObserverForEach.cpp:
* Source/WebCore/dom/InternalObserverInspect.cpp:
* Source/WebCore/dom/MessagePort.cpp:
(WebCore::MessagePort::scriptExecutionContext const):
* Source/WebCore/dom/MessagePort.h:
* Source/WebCore/dom/MutationObserver.cpp:
* Source/WebCore/dom/ScriptExecutionContext.cpp:
* Source/WebCore/dom/Subscriber.cpp:
* Source/WebCore/fileapi/FileReader.cpp:
(WebCore::FileReader::scriptExecutionContext const):
* Source/WebCore/fileapi/FileReader.h:
* Source/WebCore/html/HTMLMediaElement.cpp:
* Source/WebCore/html/ImageBitmap.cpp:
* Source/WebCore/html/OffscreenCanvas.cpp:
(WebCore::OffscreenCanvas::scriptExecutionContext const):
(WebCore::OffscreenCanvas::canvasBaseScriptExecutionContext const):
* Source/WebCore/html/OffscreenCanvas.h:
* Source/WebCore/html/VoidCallback.h:
* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
* Source/WebCore/html/closewatcher/CloseWatcher.cpp:
(WebCore::CloseWatcher::CloseWatcher):
(WebCore::CloseWatcher::scriptExecutionContext const):
* Source/WebCore/html/closewatcher/CloseWatcher.h:
* Source/WebCore/html/track/TextTrack.cpp:
(WebCore::TextTrack::scriptExecutionContext const):
* Source/WebCore/html/track/TextTrack.h:
* Source/WebCore/html/track/TrackListBase.cpp:
* Source/WebCore/inspector/InspectorInstrumentation.cpp:
(WebCore::InspectorInstrumentation::willDestroyWebGLProgram):
* Source/WebCore/inspector/InspectorInstrumentation.h:
(WebCore::InspectorInstrumentation::willDestroyWebGLProgram): Deleted.
* Source/WebCore/inspector/RTCLogsCallback.h:
* Source/WebCore/page/DOMTimer.cpp:
* Source/WebCore/page/EventSource.cpp:
(WebCore::EventSource::scriptExecutionContext const):
* Source/WebCore/page/EventSource.h:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::scriptExecutionContext const):
* Source/WebCore/page/LocalDOMWindow.h:
* Source/WebCore/page/NavigationHistoryEntry.cpp:
* Source/WebCore/page/NavigatorBase.cpp:
* Source/WebCore/page/Page.cpp:
* Source/WebCore/page/PerformanceObserver.cpp:
* Source/WebCore/page/ScreenOrientation.cpp:
(WebCore::ScreenOrientation::lock):
* Source/WebCore/page/ScreenOrientation.h:
* Source/WebCore/page/WorkerNavigator.cpp:
* Source/WebCore/testing/EventTargetForTesting.cpp:
(WebCore::EventTargetForTesting::scriptExecutionContext const):
* Source/WebCore/testing/EventTargetForTesting.h:
* Source/WebCore/testing/Internals.cpp:
* Source/WebCore/workers/Worker.cpp:
(WebCore::Worker::scriptExecutionContext const):
* Source/WebCore/workers/Worker.h:
* Source/WebCore/workers/WorkerGlobalScope.cpp:
* Source/WebCore/workers/service/ServiceWorkerContainer.cpp:
(WebCore::ServiceWorkerContainer::scriptExecutionContext const):
* Source/WebCore/workers/service/ServiceWorkerContainer.h:
* Source/WebCore/workers/service/ServiceWorkerRegistration.cpp:
* Source/WebCore/workers/shared/SharedWorker.cpp:
* Source/WebCore/workers/shared/SharedWorkerScriptLoader.cpp:
* Source/WebCore/worklets/Worklet.cpp:
* Source/WebCore/xml/XMLHttpRequest.cpp:
(WebCore::XMLHttpRequest::scriptExecutionContext const):
* Source/WebCore/xml/XMLHttpRequest.h:
* Source/WebCore/xml/XMLHttpRequestUpload.cpp:
(WebCore::XMLHttpRequestUpload::scriptExecutionContext const):
* Source/WebCore/xml/XMLHttpRequestUpload.h:

Canonical link: https://commits.webkit.org/299694@main
